File: Top Interview 150/Python/Remove_Duplicates_From_Sorted_Array_II.py
class Solution:
    def removeDuplicates(self, nums: list[int]) -> int:
        i = 0
        while i < len(nums):
            if nums.count(nums[i]) > 2:
                nums.pop(i)
            else:
                i += 1
        return len(nums)

print(Solution().removeDuplicates([1, 1, 2]))  # 3
print(Solution().removeDuplicates([0, 0, 1, 1, 1, 1, 2, 3, 3]))  # 7
print(Solution().removeDuplicates([1, 1, 1, 2, 2, 3]))  # 5

File: Top Interview 150/C++/Remove_Duplicates_From_Sorted_Array_II.cpp
#include <iostream>
#include <vector>

using namespace std;

class Solution {
public:
    int removeDuplicates(vector<int>& nums) {
        for(int i=0;i<nums.size();i++){
            if(countOccurences(nums,nums[i]) > 2){
                nums.erase(nums.begin()+i);
                i--;
            }
        }
        return nums.size();
    }
private:
    int countOccurences(vector<int>& nums,int val){
        int count = 0;
        for(int i=0;i<nums.size();i++){
            if(nums[i] == val){
                count++;
            }
        }
        return count;
    }
};

int main(){
    Solution s;
    vector<int> nums = {1,1,1,2,2,3}; // 5
    cout << s.removeDuplicates(nums) << endl;
    for(int i=0;i<nums.size();i++){
        cout << nums[i] << " ";
    }
    cout << endl;
    return 0;
}